### PR TITLE
Crash Fix for GLM-Image: TopPLogitsWarper numerical stability on MPS

### PR DESCRIPTION
**Pull request by inferencers**
# What does this PR do?
Using -inf as a filter value in masked_fill causes NaN on MPS devices.

Changing default filter_value to a large negative float -1e9 instead is a commonly used safe alternative for softmax and numerically stable across all devices (CPU, CUDA, MPS).

Fixes # (issue)
Runtime crash when generating image with GLM-Image on MPS.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/generation/logits_process.py [hunk 1]
```diff
@@ -500,7 +500,7 @@ class TopPLogitsWarper(LogitsProcessor):
     ```
     """
 
-    def __init__(self, top_p: float, filter_value: float = -float("Inf"), min_tokens_to_keep: int = 1):
+    def __init__(self, top_p: float, filter_value: float = -1e9, min_tokens_to_keep: int = 1):
```
Comment (vasqu) [inline src/transformers/generation/logits_process.py:503]: Sorry but we cannot do this as if you were combining multiple logits processors we would encounter weird behavior. Isolated this is fine but not grouped with others.

Unsure how to handle this tbh but it's certainly not good :cry:

Comment (inferencers) [reply src/transformers/generation/logits_process.py:503]: Agreed
